src/utils/media-devices-manager.js

```diff
@@ -78,6 +72,18 @@ export default class MediaDevicesManager extends EventEmitter {
     this._audioTrack = audioTrack;
   }
 
+  get defaultInputDeviceId() {
+    return this._micDevices && this._micDevices.length > 0 ? this._micDevices[0].value : NO_DEVICE_ID;
```

Comment:
I don't see a case where `this._micDevices` can ever be null since its initialized to `[]` and only assigned the result of a `Array.prototype.map` which will always return an empty array, so no need to check for this here. (Having the check made me go digging through the code trying to figure out when this state could occur, but I couldn't find it)

I think though, with the other changes this might end up only getting called once, so I would just ditch the getter altogether as the data flow will be much clearer.

Reply:
That's right, removing the check.